scripts/irods/database_interface.py

```diff
@@ -101,44 +101,41 @@ def setup_database_config(irods_config):
         'has been properly configured.\n'
         )
 
-    for k in list(irods_config.server_config.setdefault('plugin_configuration', {}).setdefault('database', {})):
-        if k != db_type:
-            del irods_config.server_config['plugin_configuration']['database'][k]
-    irods_config.server_config['plugin_configuration']['database'].setdefault(db_type, {})
```

Comment:
Tell me about the removal of the `for k in list(...):` loop.

Reply:
I think that if there were to be a loop, it would be in the upgrade script. I'm not sure in what case we would need to handle a loop at this point... If we still do need to handle a loop at this point, then something is not in line with the schema, since we don't support multiple objects in `/.../plugin_configuraiton/database` anymore.

---

Actually, instead of a loop, we probably should still do a check of `technology` to see if it matches `db_type`. We probably need to remove it if it doesn't match. This would be closer to the original behavior.

Reply:
I think I agree.

Does the schema enforce the presence of a single database element?
Is this logic only invoked during a new installation?

Reply:
> I think I agree.
> 
> Does the schema enforce the presence of a single database element?

The schema now expects:

```json
"database": {
   "details": "here"
}
```

Compared to:

```json
"database": {
   "postgres-roach-db": {
      "detail": "thing"
   }
}
```

A loop would be pretty odd here since the schema upgrade should run before, and the config is now 'flat' meaning it shouldn't be possible to have multiple configurations. Additionally, the previous schema did allow only one config option as well: https://github.com/irods/irods/blob/a9647277f3010ff5fe9123435bca68181d31b274/schemas/configuration/v5/server_config.json.in#L304-L316

> Is this logic only invoked during a new installation?

I'm not sure, just that `setup_irods.py` calls this if I recall.

Reply:
That looks good/correct to me.

We'll learn more once you start testing this PR.

Reply:
Kory and I had a brief discussion and decided that the for loop would not be necessary since:

1) It would delete configuration options, and it was decided that we shouldn't do that.
2) The current schema is flat in structure and no loop is required.

Resolving.